src/docfx/lib/log/Error.cs

```diff
@@ -56,48 +56,6 @@ public Error(ErrorLevel level, string code, FormattableString message, SourceInf
             PullRequestOnly = pullRequestOnly;
         }
 
-        public Error WithCustomRule(CustomRule customRule, bool? enabled = null)
-        {
-            var level = customRule.Severity ?? Level;
-
-            if (level != ErrorLevel.Off && customRule.ExcludeMatches(OriginalPath ?? Source?.File?.Path ?? ""))
-            {
-                level = ErrorLevel.Off;
-            }
-
-            if (enabled != null && !enabled.Value)
-            {
-                level = ErrorLevel.Off;
-            }
-
-            var message = Message;
-
-            if (!string.IsNullOrEmpty(customRule.Message))
-            {
-                try
-                {
-                    message = string.Format(customRule.Message, MessageArguments);
-                }
-                catch (FormatException)
-                {
-                    message += "ERROR: custom message format is invalid, e.g., too many parameters {n}.";
-                }
-            }
-
-            message = string.IsNullOrEmpty(customRule.AdditionalMessage) ?
-                message : $"{message}{(message.EndsWith('.') ? "" : ".")} {customRule.AdditionalMessage}";
-
-            return new Error(
-                level,
-                string.IsNullOrEmpty(customRule.Code) ? Code : customRule.Code,
-                message,
-                MessageArguments,
-                Source,
-                PropertyPath,
-                OriginalPath,
-                customRule.PullRequestOnly);
-        }
-
         public Error WithLevel(ErrorLevel level)
         {
             return level == Level ? this : new Error(level, Code, Message, MessageArguments, Source, PropertyPath, OriginalPath, PullRequestOnly);
```

Comment:
Maybe use [FormattableStringFactory](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.formattablestringfactory?view=netcore-3.1) so we can only have one constructor for Error and remove MessageArguments from that class?

Reply:
looks it will make caller more complex